File: My Practice Projects/PracticeListView/app/src/main/java/com/example/farhan/practicelistview/CustomAdapter.java
package com.example.farhan.practicelistview;


import android.content.Context;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import java.util.ArrayList;

public class CustomAdapter extends ArrayAdapter {

    private ArrayList<DataSource> mDataSoruce;

    public CustomAdapter(@NonNull Context context, ArrayList<DataSource> mDataSoruce ) {
        super(context,0,mDataSoruce );
        this.mDataSoruce = mDataSoruce;
    }

    @NonNull
    @Override
    public View getView(final int position, @Nullable View convertView, @NonNull ViewGroup parent) {

        View view;

        if(convertView == null){

            view = LayoutInflater.from(getContext()).inflate(R.layout.customlistview,parent,false);
        }else {
            view = convertView;
        }
        DataSource ds = mDataSoruce.get(position);

        ImageView imageView = (ImageView)  view.findViewById(R.id.mImage);
        TextView textView1 = (TextView)  view.findViewById(R.id.mTextView1);
        TextView textView2 = (TextView) view.findViewById(R.id.mTextView2);
        Button button =  view.findViewById(R.id.mbtn);

        imageView.setImageResource(ds.getImage());
        textView1.setText(String.valueOf(ds.getId()));
        textView2.setText(ds.getName());

        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Toast.makeText(getContext(), "Button Clicked: "+position, Toast.LENGTH_SHORT).show();

            }
        });

        return view;
    }
}

File: My Practice Projects/PracticeKotlinRecyclearView/app/src/main/java/com/example/farhan/practicekotlinrecyclearview/CustomAdapter.kt
package com.example.farhan.practicekotlinrecyclearview

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

class CustomAdapter(val data: ArrayList<String>) : RecyclerView.Adapter<CustomAdapter.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        return ViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.reservation_form_b_item_view, parent, false))
    }

    override fun getItemCount(): Int = data.size

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.bindView(data[position])
    }


    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        fun bindView(obj: String) {

        }
    }


}